src/clients/victoriaMetrics/metricFunctions.ts

```diff
@@ -0,0 +1,43 @@
+export function generateLabelCombinations(
+  labelOptions: Record<string, Set<string>>,
+): Record<string, string>[] {
+  const labelNames = Object.keys(labelOptions);
+  const labelCombinations: Record<string, string>[] = [];
+
+  function buildCombinations(
+    currentCombination: Record<string, string>,
+    index: number,
+  ) {
+    if (index === labelNames.length) {
+      labelCombinations.push(currentCombination);
+      return;
+    }
+
+    const key = labelNames[index];
+    const options = Array.from(labelOptions[key]);
+
+    options.forEach((option) => {
+      buildCombinations({ ...currentCombination, [key]: option }, index + 1);
+    });
+  }
+
+  buildCombinations({}, 0);
+
+  return labelCombinations;
+
+  // TODO-kev: Use above recursive solution or below compact one
```

Comment:
Let's use the non-recursive, compact function, as long as it conforms to ur unit tests.